src/spoke/interfaces/ISpoke.sol

```diff
@@ -11,22 +11,27 @@ import {IShareToken} from "src/spoke/interfaces/IShareToken.sol";
 import {IVaultFactory} from "src/spoke/factories/interfaces/IVaultFactory.sol";
 import {IVault, VaultKind} from "src/spoke/interfaces/IVault.sol";
 import {Price} from "src/spoke/types/Price.sol";
+import {IRequestManager} from "src/spoke/interfaces/IRequestManager.sol";
 
 /// @dev Centrifuge pools
 struct Pool {
     uint256 createdAt;
     mapping(ShareClassId scId => ShareClassDetails) shareClasses;
 }
 
+struct ShareClassAsset {
+    /// @dev Manager that can send requests, and handles the request callbacks.
+    IRequestManager manager;
+    /// @dev Number of linked vaults.
+    uint32 numVaults;
```

Comment:
What is the reason we have this counter?

Reply:
To block updating request manager if there is >0 linked vault.